src/Week4_HW.py

```diff
@@ -18,51 +20,105 @@ def set_conf():
     return conf
 
 
-def main():
+def read_data(table_name):
+    """
+    Purpose: Function is to grab data from a user specified table from baseball database
 
-    # Making Spark Object
+    """
+    # Creating pyspark object
     sc = spark.SparkContext.getOrCreate(conf=set_conf())
     sqlContext = SQLContext(sc)
 
-    """
-    Different method to connect sql database
-
-    spark = SparkSession \
-        .builder \
-        .appName("App") \
-        .master('local[*]') \
-        .config('spark.jars', '/home/bioinfo/Desktop/mysql-connector-java-5.1.46/mysql-connector-java-5.1.46.jar') \
-        .enableHiveSupport() \
-        .getOrCreate() \
-    """
-
-    # Making connection to MariaDB
+    # Making connection to mariadb
     mysql_db_driver_class = "com.mysql.jdbc.Driver"
     host_name = "localhost"
     port_no = "3306"
     user_name = "root"
     password = "Hamid&Mahasty1"  # pragma: allowlist secret
     database_name = "baseball"
 
-    # test_query = """
-    # SELECT * FROM batter_counts
-    # """
-
     # Making JDBC URL
-    mysql_jdbc_url = "jdbc:mysql://" + host_name + ":" + port_no + "/" + database_name
+    mysql_jdbc_url = (
+        "jdbc:mysql://"
+        + host_name
+        + ":"
+        + port_no
+        + "/"
+        + database_name
+        + "?zeroDateTimeBehavior=convertToNull"  # to turn 0 timestamp into null
+    )
 
-    # Reading DataTable from jdbc
-    jdbcDF = (
+    sql_data = (
         sqlContext.read.format("jdbc")
         .option("url", mysql_jdbc_url)
         .option("driver", mysql_db_driver_class)
-        .option("dbtable", "batter_counts")
+        .option("dbtable", table_name)
         .option("user", user_name)
         .option("password", password)
         .load()
     )
+    return sql_data
+
+
+def main():
+
+    # Creating pyspark object
+    sc = spark.SparkContext.getOrCreate(conf=set_conf())
+    sqlContext = SQLContext(sc)
+
+    # Reading DataTable from jdbc
+    batter_counts_rdd = read_data("batter_counts")
+    game_rdd = read_data("game")
+
+    # Created Pyspark table for queries
+    batter_counts_rdd.createOrReplaceTempView("batter_counts")
+    batter_counts_rdd.persist(StorageLevel.DISK_ONLY)
+    game_rdd.createOrReplaceTempView("game")
+    game_rdd.persist(StorageLevel.DISK_ONLY)
+
+    # Creating a temporary table
+    tempTableDf = sqlContext.sql(
+        """
+    SELECT bc.game_id, bc.batter, atBat, Hit,
+    team_id, local_date
+    FROM batter_counts bc
+    JOIN game g
+    ON bc.game_id = g.game_id;
+        """
+    )
+
+    tempTableDf.createOrReplaceTempView("temp")
+    tempTableDf.persist(StorageLevel.DISK_ONLY)
+
+    # Creating Rolling averag
+    rolling_avg_df = sqlContext.sql(
+        """
+    SELECT t.batter, CAST(t.local_date AS VARCHAR(30)),
```

Comment:
might be helful to seperate individual selections in the select statement...better readability